upgrader.go

```diff
@@ -213,6 +214,37 @@ func (u *Upgrader) Upgrade() error {
 	}
 }
 
+// WaitForParentExit wait until parent exit if parent exists
+// otherwise returns directly
+// returns nil when OK, non nil when NG
+func (u *Upgrader) WaitForParentExit(ctx context.Context) error {
+	if u.parent == nil {
+		return nil
+	}
+
+	if u.parentErr != nil {
```

Comment:
This introduces a race condition, since `parentErr` is written under the semaphore.